source/adios2/toolkit/derived/Function.cpp

```diff
@@ -39,6 +39,23 @@ T *ApplyOneToOne(Iterator inputBegin, Iterator inputEnd, size_t dataSize,
     return outValues;
 }
 
+template <class T>
+T *ApplyOneToOneOnce(T *inputData, size_t dataSize, std::function<T(T)> compFct)
+{
+    T *outValues = (T *)malloc(dataSize * sizeof(T));
+    if (outValues == nullptr)
+    {
+        helper::Throw<std::invalid_argument>("Derived", "Function", "ApplyOneToOneOnce",
+                                             "Error allocating memory for the derived variable");
+    }
+    for (size_t i = 0; i < dataSize; i++)
+    {
+        T data = *(reinterpret_cast<T *>(inputData + i));
+        outValues[i] = compFct(data);
+    }
+    return outValues;
+}
+
```

Comment:
This function is basically a `std::transform`. Check how sqrt is implemented now and update the trig functions using the same method.